src/main/webapp/script.js

```diff
@@ -0,0 +1,25 @@
+var clicked = false;
+
+function getVideoResults() {
```

Comment:
Up to you, I know this syntax looks a little verbose.

But you can modify this function to an async function just to make this a little bit more readable:
https://developers.google.com/web/fundamentals/primers/async-functions

Reply:
I'll look into it more.